pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -116,6 +116,24 @@ func (r *ReconcileMasterUserRecord) Reconcile(request reconcile.Request) (reconc
 			logger.Error(err, "unable to add finalizer to MasterUserRecord")
 			return reconcile.Result{}, err
 		}
+		// migration for CRT-1075: add an annotation with the email address (same as for associated UserSignup resource)
+		usersignup := &toolchainv1alpha1.UserSignup{}
+		if r.client.Get(context.TODO(), types.NamespacedName{
+			Namespace: mur.Namespace,
+			Name:      mur.Spec.UserID,
+		}, usersignup); err != nil {
+			logger.Error(err, "unable to get associated UserSignup")
+			return reconcile.Result{}, err
+		}
+		if mur.Annotations == nil {
+			mur.Annotations = map[string]string{}
+		}
+		mur.Annotations[toolchainv1alpha1.MasterUserRecordEmailAnnotationKey] = usersignup.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey]
```

Comment:
Check if the annotation is not already set before executing the migration step.

Reply:
good point! done in 646e3f1